patches/0027-Do-not-compute-tags_all-at-TF-level.patch

```diff
@@ -336,21 +203,185 @@ index 0bc956a747..e0636e99ce 100644
  	return nil
  }
  
+diff --git a/shim/pf.go b/shim/pf.go
+new file mode 100644
+index 0000000000..65f8ffaeb5
+--- /dev/null
++++ b/shim/pf.go
+@@ -0,0 +1,142 @@
++package shim
++
++import (
++	"context"
++
++	"github.com/hashicorp/terraform-plugin-framework/function"
++	"github.com/hashicorp/terraform-plugin-framework/provider"
++	"github.com/hashicorp/terraform-plugin-framework/resource"
++	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
++)
++
++type transformedFrameworkProvider struct {
```

Comment:
There is an interesting Go problem here I'd love eyes on cc @blampe  - the intent is to override the response of provider.Provider interface at some depth that is all I'd like to do is change this response :

```
provider.Resources(ctx)[17 /*arbitrary index*/]().Schema(ctx, request, response)
```

Applying the technique of embedding interfaces into structs and overriding.

However..

There is an open set of extra interfaces like resource.ResourceWithModifyPlan though that the framework might be probing for. 

The danger with struct wrappers it would seem is that the wrapped provider or resource will no longer pass a type test on these interfaces and possibly lead to invalid behavior? I'm implementing all these interfaces here that I could find, but there still is the risk that new interfaces will be added upstream and this code will need updates and fail quietly. Is there anything better we can do?

Reply:
I'm a bit worried about adding this without running the full set of upstream tests.
I don't think we have enough coverage in our existing tests to catch potential problems here.